### ISSUE TITLE
Performance on hard distribution mode

### ISSUE THREAD

**Issue by KarlXing** [lerrytang/train-procgen-pfrl#1]
Hi, thank you for sharing this. It's nice to see the effort of training procgen tasks with Pytorch. I wonder if you have tested the code with hard levels. I also tried to use PPO to train procgen tasks with my own Pytorch code. The code works fine with easy levels. However, my code was not able to achieve similar performance as the original Tensorflow implementation with hard levels even all hyperparameters are same. Just curious if you meet similar problems :)

**Comment by lerrytang** [comment]
Hi! No, I haven't tried the hard levels.
Can you give me a specific game that you found the score deviated a lot from the paper? I'll launch a job and see the results.

**Comment by KarlXing** [comment]
That's very kind of you! I appreciate it if you could test dodgeball and starpilot with 500 levels of generalization :)

**Comment by lerrytang** [comment]
![Screen Shot 2021-04-06 at 20 45 50](https://user-images.githubusercontent.com/2889514/113705984-13f91680-9719-11eb-8ad8-ea644f81c23d.png)
This is the results from one single run.  
Comparing with Fig. 4 in the paper, starpilot is quite different while dodgeball is more or less close to what is reported.
Is this consistent with your results?

**Comment by KarlXing** [comment]
Thank you for sharing the result! I think that's consistent with my results if I use batch normalization in the IMPALA model. Without batch normalization, my results could be worse. I will look into more details in your implementation and probably also the official version to figure out the cause of the performance gap. Once I find anything, I will let you know!
